docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,798 @@
+and denies requests from old versions of "web".
+
+```yaml
+type: MeshTrafficPermission
+mesh: mesh-1
+name: backend-permissions
+targetRef:
+  kind: MeshService
+  name: backend
+from:
+  - targetRef:
+      kind: MeshServiceSubset
+      name: web
+      tags:
+        version: v1
+    conf:
+      action: DENY
+  - targetRef:
+      kind: Mesh
+    conf:
+      action: ALLOW
+```
+
+Now we want to figure
+what configuration we'll get on "backend" data plane proxy.
+
+Merge all "from" sections from all policies
+matched for "backend" proxy:
+
+```yaml
+from:
+  - targetRef: # from backend-permissions
+      kind: MeshServiceSubset
+      name: web
+      tags:
+        version: v1
+    conf:
+      action: DENY
+  - targetRef: # from backend-permissions
+      kind: Mesh
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-monitoring
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-logger
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: Mesh
+    conf:
+      action: DENY
+```
+
+Now we'll look at each possible traffic source individually.
+
+Select "from" sections that target "web" with "v1":
+
+```yaml
+from:
+  - targetRef: # from backend-permissions
+      kind: MeshServiceSubset
+      name: web
+      tags:
+        version: v1
+    conf:
+      action: DENY
+  - targetRef: # from backend-permissions
+      kind: Mesh
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: Mesh
+    conf:
+      action: DENY
+```
+
+and merge into single configuration:
+
+```yaml
+conf:
+  action: DENY # from backend-permissions
+```
+
+Select "from" sections that target "infra-monitoring":
+
+```yaml
+from:
+  - targetRef: # from backend-permissions
+      kind: Mesh
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-monitoring
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: Mesh
+    conf:
+      action: DENY
+```
+
+and merge into single configuration:
+
+```yaml
+conf:
+  action: ALLOW # from backend-permissions
+```
+
+Select "from" sections that target "infra-logger":
+
+```yaml
+from:
+  - targetRef: # from backend-permissions
+      kind: Mesh
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-logger
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: Mesh
+    conf:
+      action: DENY
+```
+
+and merge into single configuration:
+
+```yaml
+conf:
+  action: ALLOW # from backend-permissions
+```
+
+#### ProxyTemplate
+
+ProxyTemplate is neither inbound nor outbound policy, 
+so it doesn't have "from" or "to" arrays. 
+There are only "imports" and "modifications" arrays
+which override arrays from other levels.
```

Comment:
I think we should consider appending arrays or introducing mechanism of composing proxy template. Imagine you have Mesh level ProxyTemplate with 5 modifications (because again, they are not composable). Then you want to do something for one service, you need to copy paste those 5 modifications from Mesh level to not break anything.